HintList.py

```diff
@@ -215,7 +215,7 @@ def tokens_required_by_settings(world: World) -> int:
 # Hints required under certain settings
 conditional_always: dict[str, Callable[[World], bool]] = {
     'Market 10 Big Poes':           lambda world: world.settings.big_poe_count > 3 and 'big_poes' not in world.settings.misc_hints,
-    'Deku Theater Mask of Truth':   lambda world: not world.settings.complete_mask_quest and 'Mask of Truth' not in world.settings.shuffle_child_trade,
+    'Deku Theater Mask of Truth':   lambda world: (not world.settings.complete_mask_quest and 'Mask of Truth' not in world.settings.shuffle_child_trade) and not ('mask_of_truth' not in world.settings.misc_hints or 'mask_shop_dual' not in world.settings.misc_hints),
```

Comment:
Any particular reason for the double negative here? If not, I'd suggest removing it to make the code easier to read.